### ISSUE TITLE
Do the tiles fit seamlessly?

### ISSUE THREAD

**Issue by bootchk** [BorisTheBrave/resynth-tiles#2]
I am the maintainer of Resynthesizer.  I am intrigued by your plugin.  Thanks.

Please pardon my ignorance, but do the tiles (that your plugin produces) fit toegether seamlessly (without any hard edges) or does that depend on further processing by say the app that consumes the tiles?

**Comment by BorisTheBrave** [comment]
The tiles mutally fit together seamlessly at the edges. This is the value added by my plugin brings vs just running Resynthesizer itself.

The example from https://github.com/BorisTheBrave/resynth-tiles/wiki/Tutorial has no processing done to it except Resynthesizer. The tiles are directly placed next to each other using Tiled with no margin or blending.

**Comment by bootchk** [comment]
Thanks.  I do not think Resynthesizer alone would makes the tiles fit seamlessly with each other (only each tile seamless with itself.) It helps me to know that there is something in your code that does that, I can read your code more carefully.

I also know that Paul Harrison has an example where he applies texture to a "logo", a text, which is what your tutorial accomplishes in a different way.  I suppose having a tile set is more general and has other uses.  An interesting idea.

Paul Harrison (on his website) also had a tile generator which you might find interesting.

**Comment by BorisTheBrave** [comment]
> Paul Harrison (on his website) also had a tile generator which you might find interesting.

What is the link?

> It helps me to know that there is something in your code that does that, I can read your code more carefully.

This plugin works by invoking resynthesizer multiple times, with different border or wrapping. It's done in such away so that early tiles act as the border for later tiles, forcing them to be consistent.

**Comment by bootchk** [comment]
http://www.logarithmic.net/pfh/ghost-diagrams

I hacked at it also, I recall I was generating his kind of tiles brute force, then testing whether a set would tile the plane in a pleasing way.  Probably like you, too many interesting things and not enough time to do them.

**Comment by BorisTheBrave** [comment]
Yes, I've done lots of tile investigations, and now have a [commercial product ](https://assetstore.unity.com/packages/tools/modeling/tessera-pro-161077)based off WFC, which is not dissimilar from that link.

**Comment by bootchk** [comment]
Very cool.  So, a plugin for the Unity app?

I have also dabbled in commericial apps for iOS, never with success.  Marketing is hard.  I hope you have found a niche (game builders?) where word of mouth works well.

Someone else is marketing "heal selection" as a standalone app, https://en.wikipedia.org/wiki/Inpaint.  I suppose I should have.

Your big tiles is something like what I was trying: each tile has a sequence of "colors" on a side, you match sequences.  Generating tilesets at random.  Then a backtracking algorithm to see whether they would tile a plane.

I also played with generating graphs and organic molecules . Following valence rules, using a language like BNF for describing transformations, and generating sentences from it, where the sentences canonically describe molecules in what is called the SMILES notation, which is an ASCII string.

Might be of interest:   https://jmvalin.ca/daala/paint_demo/  about an "intra paint algorithm" possibly used in MPEG encoding?
They divide an image into tiles, keep the sequence of colors along the edge (shared between two tiles), calculate a gradient or direction in each tile, and then repaint each tile somehow.  I did not discover a reference giving enough how to implement it.
